Add configurable loading screen.

dmarcos opened the pull request:
**Description:**

Loading screen to avoid blank screen while loading assets and scene render loop has not yet started. The loader animation doesn't require 60fps to mask main thread pauses.

The logic is a `pseudo-component` since it has to initialize and render while the assets load. 

The title of the page is displayed above the loading indicator.

The color of the dots, background and style of the title bar are configurable.

![loader](https://d3vv6lp55qjaqc.cloudfront.net/items/0M3q3m0W2Z3h3G17152y/Image%202018-09-05%20at%203.06.05%20AM.png?X-CloudApp-Visitor-Id=222920&v=51eab09b)

ngokevin commented on src/core/scene/a-scene.js:
```diff
@@ -507,6 +510,80 @@ module.exports.AScene = registerElement('a-scene', {
       writable: true
     },
 
+    setupLoader: {
```
Can place all the loader code in a module that exports functions `setupLoader(sceneEl)` to not pollute scope, not meant to be callable by user

ngokevin commented on src/core/scene/a-scene.js:
```diff
@@ -507,6 +510,80 @@ module.exports.AScene = registerElement('a-scene', {
 
+    setupLoader: {
+      value: function () {
+        var self = this;
```
a little daunting to read a web of instantiations that aren't used until later in various places. need to jump up/down the function to follow.

ngokevin commented on src/core/scene/a-scene.js:
```diff
@@ -507,6 +510,80 @@ module.exports.AScene = registerElement('a-scene', {
+      writable: true
+    },
+
+    removeLoader: {
```
Can also free the created geometry/materials

dmarcos commented:
@ngokevin This is ready to go:

- Extracted `loadingScreen` logic to module.
- Added tests.
- Now the loading screen also works on vr if `vrdisplayactivate` fires. A-Frame will always render something on link traversal while the scene is loading (unless the user disables the loader).

ngokevin commented:
I'd document the `document.title` behavior
